Source/WebCore/page/Location.cpp

```diff
@@ -64,11 +65,11 @@ const Frame* Location::frame() const
 
 const URL& Location::url() const
 {
-    RefPtr localFrame = dynamicDowncast<LocalFrame>(frame());
-    if (!localFrame)
-        return aboutBlankURL();
+    RefPtr localWindow = dynamicDowncast<LocalDOMWindow>(*m_window);
+    if (!localWindow)
+        return *new URL(emptyString());
```

Comment:
should be `return aboutBlankURL();`

Reply:
You're leaking memory here.

Reply:
> should be `return aboutBlankURL();`

As mentioned at https://github.com/WebKit/WebKit/pull/18026#discussion_r1354620288, returning `aboutBlankURL()` if the window (or frame) is null isn’t spec-conformant — it causes the _“document.open() does not change document's URL (active but not fully active document)”_ test case at
https://wpt.fyi/results/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.html
to fail. So it’s unclear to me what otherwise we can return here.

Reply:
Then something like this:
```
if (!localWindow) {
    static NeverDestroyed<URL> emptyURL = emptyString();
    return emptyURL.get();
}
```

Your current proposal is leaking memory, which is what I am objecting to.

If this works, this would be better too:
```
if (!localWindow) {
    static NeverDestroyed<URL> nullURL;
    return nullURL.get();
}
```

Reply:
> If this works, this would be better too:
> 
> ```
> if (!localWindow) {
>     static NeverDestroyed<URL> nullURL;
>     return nullURL.get();
> }
> ```

Thanks — that works. Pushed now now